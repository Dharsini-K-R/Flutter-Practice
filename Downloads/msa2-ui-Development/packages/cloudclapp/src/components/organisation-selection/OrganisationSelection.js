import React, { useCallback, useRef } from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import useToggle from "react-use/lib/useToggle";

import { makeStyles, Button, Popover } from "@material-ui/core";
import SelectSearch from "cloudclapp/src/components/select-search/SelectSearch";
import {ExchangeAltIcon} from "react-line-awesome";

const useStyles = makeStyles((theme) => {
  const { palette, breakpoints, typography } = theme;
  return {
    topNav: {
      display: "flex",
      [breakpoints.down("md")]: {
        margin: 10,
        order: 2,
      },
    },
    breadcrumbContainer: {
      display: "flex",
    },
    title: {
      fontSize: "1rem",
      letterSpacing: 0.5,
      lineHeight: "1.8rem",
      maxWidth: 200,
    },
    btnRoot: {
      height: "auto",
      marginRight: "-25px",
      background: "transparent",
      borderRadius: 4,
      fontSize: "0.875rem",
      fontWeight: typography.fontWeightMedium,
      lineHeight: 1,
      letterSpacing: 0.3,
      border: 0,
      "& .MuiButton-label": {
        textTransform: "none",
      },
    },
    paper: {
      overflowY: "inherit",
      overflowX: "inherit",
      backgroundColor: "transparent",
    },
    popoverContainer: {
      position: "relative",
      top: 7,
      width: 300,
      flexGrow: 1,
      boxShadow:
        "0 6px 16px 4px rgba(81, 97, 133, 0.13), 0 3px 8px 0 rgba(0, 0, 0, 0.15)",
      border: "1px solid" + palette.border.greyLight1,
      borderRadius: "5px 5px 0 0",
      backgroundColor: palette.background.paper,
      "&:after, &:before": {
        bottom: "100%",
        left: "50%",
        border: "solid transparent",
        content: '" "',
        height: 0,
        width: 0,
        position: "absolute",
        pointerEvents: "none",
      },
      "&:after": {
        borderColor: "rgba(255, 255, 255, 0)",
        borderBottomColor: "#fff",
        borderWidth: 6,
        marginLeft: -6,
      },
      "&:before": {
        borderColor: "rgba(191, 201, 217, 0)",
        borderBottomColor: palette.border.greyLight1,
        borderWidth: 7,
        marginLeft: -7,
      },
      OrgSelectIcon:{
        background: "#46587E",
      },
    },
  };
});

const OrganisationSelection = ({
  tenants,
  multiOrg,
  onSelectOrganisation,
}) => {
  const { t } = useTranslation();
  const classes = useStyles();
  const organisationButton = useRef(null);
  const [organisationIsOpen, toggleOrganisation] = useToggle(false);

  const dismissPopovers = useCallback(() => {
    toggleOrganisation(false);
  }, [toggleOrganisation]);

  return (
    <nav className={classes.topNav}>
      {/* Organisation Selection */}
      <div className={classes.breadcrumbContainer}>
        {multiOrg ? (
          <Button
            id="cclaOrganisationBtn"
            aria-haspopup="true"
            variant="outlined"
            aria-label="organisation"
            size="small"
            disableRipple
            onClick={toggleOrganisation}
            classes={{
              root: classes.btnRoot,
              label: classes.btnLabel,
            }}
            className={organisationIsOpen ? classes.btnOpen : classes.btnClosed}
            ref={organisationButton}
          >
            <ExchangeAltIcon className={classes.OrgSelectIcon}/>
            <Popover
              id="cclaorganisationselection"
              open={organisationIsOpen}
              anchorEl={organisationButton.current}
              onClose={dismissPopovers}
              BackdropComponent={() => <></>}
              classes={{
                paper: classes.paper,
              }}
              anchorOrigin={{
                vertical: "bottom",
                horizontal: "center",
              }}
              transformOrigin={{
                vertical: "top",
                horizontal: "center",
              }}
            >
              <div
                id="cclaOrganisationSelectSearchDiv"
                className={classes.popoverContainer}
              >
                <SelectSearch
                  id="cclaorganisation"
                  options={tenants}
                  noOptionText={t("Organisation not found.")}
                  isOrgSelection={true}
                  onSelect={onSelectOrganisation}
                  isClearable={false}
                />
              </div>
            </Popover>
          </Button>
        ) : (
          null
        )}
      </div>
    </nav>
  );
};

OrganisationSelection.propTypes = {
  tenants: PropTypes.array.isRequired,
  multiOrg: PropTypes.bool.isRequired,
  onSelectOrganisation: PropTypes.func.isRequired,
};

export default OrganisationSelection;
