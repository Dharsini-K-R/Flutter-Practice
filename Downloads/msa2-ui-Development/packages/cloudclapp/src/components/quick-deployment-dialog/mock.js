export const cloudConnectionSummary = [
  {
    "cloudVendor": "aws",
    "cloudType": "Public Cloud",
    "vendorDisplayName": "AWS",
    "connections": [
      {
        "connectionStatus": "",
        "connectionName": "aws-connection-1"
      }
    ]
  },
  {
    "cloudVendor": "azure",
    "cloudType": "Public Cloud",
    "vendorDisplayName": "AZURE",
    "connections": [
      {
        "connectionStatus": "",
        "connectionName": ""
      }
    ]
  },
  {
    "cloudVendor": "gcp",
    "cloudType": "Public Cloud",
    "vendorDisplayName": "Google Cloud Platform",
    "connections": [
      {
        "connectionStatus": "",
        "connectionName": "gcp-connection-1"
      }
    ]
  }
]