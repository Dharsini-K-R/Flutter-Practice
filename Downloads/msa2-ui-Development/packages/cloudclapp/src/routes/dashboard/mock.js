export const organisation = [
  { environmentCount: 20, usersCount: 1, prefix: "00p" },
];

export const environments = [
  {
    status: "Critical",
    version: "V1.1.1",
    user: "Bib Dev",
    env: "Development",
    envData:
      "Development Environment for all new projects and major updates. Nginx and PHP added to the Environment",
    deplymentCount: "2",
    cost: "58.2",
  },
];

export const cloudVendors = {
  aws: {
    displayName: "AWS",
    category: "Public Cloud",
    logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/aws.logo",
    credentials_required: ["aws_access_key", "aws_secret_key"],
    services: {
      eks: {
        displayName: "Elastic Container Service for Kubernetes",
        logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/aws_eks.logo",
        workflow: {
          env:
            "Process/cloudclapp-wf/Provision_an_EKS_Cluster__AWS_/Provision_an_EKS_Cluster__AWS_",
          appsDeployment:
            "Process/cloudclapp-wf/Provision_Apps_EKS_Cluster/Provision_Apps_EKS_Cluster",
          scanApp:
            "Process/cloudclapp-wf/Scan_a_App_Container_/Scan_a_App_Container_",
        },
      },
    },
  },
  azure: {
    displayName: "AZURE",
    category: "Public Cloud",
    logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/azure.logo",
    credentials_required: ["azure_client_id", "azure_client_secret"],
    services: {
      aks: {
        displayName: "Azure Kubernetes Service",
        logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/azure_aks.logo",
        workflow: {
          env:
            "Process/cloudclapp-wf/Provision_an_AKS_Cluster__AZURE_/Provision_an_AKS_Cluster__AZURE_",
          appsDeployment:
            "Process/cloudclapp-wf/Provision_an_AKS_Apps__AZURE_/Provision_an_AKS_Apps__AZURE_",
          scanApp:
            "Process/cloudclapp-wf/Scan_a_App_Container_/Scan_a_App_Container_",
        },
      },
    },
  },
  gcp: {
    displayName: "Google Cloud Platform",
    category: "Public Cloud",
    logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/gcp.logo",
    credentials_required: [
      "gcp_service_account_email",
      "gcp_private_key_id",
      "gcp_private_key",
      "gcp_client_id",
    ],
    services: {
      gke: {
        displayName: "Google Kubernetes Engine",
        logo: "Process/cloudclapp-wf/CCLA_Adapters/logos/gcp_gke.logo",
        workflow: {
          env:
            "Process/cloudclapp-wf/Provision_an_GKE_Cluster__GCP_/Provision_an_GKE_Cluster__GCP_",
          appsDeployment:
            "Process/cloudclapp-wf/Provision_Apps_GKE_Cluster/Provision_Apps_GKE_Cluster",
          scanApp:
            "Process/cloudclapp-wf/Scan_a_App_Container_/Scan_a_App_Container_",
        },
      },
    },
  },
};
