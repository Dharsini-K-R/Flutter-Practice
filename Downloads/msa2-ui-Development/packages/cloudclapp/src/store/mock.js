export const tenantList = [
  {
    value: "ARA",
    id: 15,
    ubiqubeId: null,
    label: "Amit",
  },
  {
    value: "bfR",
    id: 28,
    ubiqubeId: null,
    label: "bfR",
  },
  {
    value: "fbo",
    id: 11,
    ubiqubeId: null,
    label: "fbo_tenant",
  },
  {
    value: "FUJ",
    id: 12,
    ubiqubeId: null,
    label: "Fujitsu",
  },
  {
    value: "hto",
    id: 69,
    ubiqubeId: null,
    label: "Htoper",
  },
  {
    value: "PRN",
    id: 2,
    ubiqubeId: null,
    label: "PRNOperator",
  },
  {
    value: "MSA",
    id: 10,
    ubiqubeId: null,
    label: "UBIqube",
  },
];

export const userDetails = {
  id: 1,
  baseRole: { id: 1, name: "Super Administrateur" },
  externalReference: "NCLG1",
  login: "ncroot",
  operator: {
    id: 1,
    contacts: [],
    abonnes: [],
    baseUrl: null,
    isPartner: false,
    name: "UBIqube",
    prefix: "NCL",
    address: {
      city: "",
      country: "",
      fax: "",
      mail: "",
      phone: "",
      streetName1: "",
      streetName2: "",
      streetName3: "",
      zipCode: "",
    },
    sort: "NAME",
    companyType: "",
  },
  pwd: "",
  userType: 2,
  acteurId: 1,
  address: {
    city: "Dublin",
    country: "Ireland",
    fax: "111111",
    mail: "support@ubiqube.com",
    phone: "10000000000",
    streetName1: "Some street address1",
    streetName2: "Some street address2",
    streetName3: "Some street address3",
    zipCode: "000000",
  },
  delegationProfileId: 0,
  delegations: null,
  delegationsPerSubtenant: {},
  netceloId: {
    ubiID: "NCLG1",
    thePrefix: "NCL",
    id: 1,
    customerPrefix: "NCL",
  },
  firstname: "Clark",
  manageAllUsers: true,
  name: "ncroot",
  sort: "NAME",
  ldapAuthentication: false,
  delegationProfilePerCustomer: {},
};

export const loginSuccess = {
  message: "authenticated",
  userDetails,
  token:
    "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJuY3Jvb3QiLCJpYXQiOjE1NjIxNjEwNDQsImx2bCI6IjEiLCJleHAiOjE1NjIyNDc0NDR9.fDYfGQqJpWgI0iGtwuUWTFXSa2T51JlSWXYPkrx2QgwBy3vPfKeD6xc0RrH2ekZdh2rAucCUclunNl1tXBCwxw",
};
