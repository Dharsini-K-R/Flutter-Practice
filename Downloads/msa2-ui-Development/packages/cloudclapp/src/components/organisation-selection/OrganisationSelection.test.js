import React from "react";
import {
  renderWithProvider,
  fireEvent,
  waitForElement,
} from "cloudclapp/src/utils/test-utils";
import "@testing-library/jest-dom/extend-expect";
import OrganisationSelection from "./OrganisationSelection";
import { tenantList } from "cloudclapp/src/store/mock.js";

const mockState = {
  designations: tenantList,
};

jest.mock("react-select", () => ({ options, value, onChange }) => {
  const handleChange = (event) => {
    const option = options.find(
      (option) => option.label === event.currentTarget.value,
    );
    onChange(option);
  };
  return (
    <>
      {/* eslint-disable-next-line jsx-a11y/no-onchange */}
      <select data-testid="select" value={value} onChange={handleChange}>
        {options.map(({ id, label }) => (
          <option key={id} value={label}>
            {label}
          </option>
        ))}
      </select>
    </>
  );
});

const blankDesignation = {
  value: null,
  id: null,
  label: null,
  ubiqubeId: null,
};

const render = (childComponent) =>
  renderWithProvider({ childComponent, mockState });

describe("OrganisationSelection component", () => {
  it("renders correctly", () => {
    const mockSelectOrganisation = jest.fn();
    const { asFragment } = render(
      <OrganisationSelection
        multiOrg={true}
        tenants={tenantList}
        onSelectOrganisation={mockSelectOrganisation}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("handles tenant selection correctly", async () => {
    const mockSelectOrganisation = jest.fn();
    const { getByLabelText, getByTestId } = render(
      <OrganisationSelection
        multiOrg={true}
        tenants={tenantList}
        onSelectOrganisation={mockSelectOrganisation}
      />,
    );
    const tenantLabel = getByLabelText("organisation");
    fireEvent.click(tenantLabel);
    const select = await waitForElement(() => getByTestId("select"));
    fireEvent.change(select, { target: { value: "Amit" } });
    expect(mockSelectOrganisation.mock.calls.length).toBe(1);
    expect(mockSelectOrganisation.mock.calls[0][1].label).toBe("Amit");
  });

  it("handles displaying selected tenant correctly", () => {
    const mockSelectOrganisation = jest.fn();
    const { getByLabelText } = render(
      <OrganisationSelection
        multiOrg={true}
        tenants={tenantList}
        onSelectOrganisation={mockSelectOrganisation}
      />,
    );
    const tenantLabel = getByLabelText("organisation");
    expect(tenantLabel).toBeDefined();
  });
});
