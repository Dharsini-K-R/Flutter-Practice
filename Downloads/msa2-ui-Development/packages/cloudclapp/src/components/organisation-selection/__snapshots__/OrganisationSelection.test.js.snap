// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganisationSelection component renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="makeStyles-topNav"
  >
    <div
      class="makeStyles-breadcrumbContainer"
    >
      <button
        aria-haspopup="true"
        aria-label="organisation"
        class="MuiButtonBase-root MuiButton-root makeStyles-btnRoot MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
        id="cclaOrganisationBtn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <i
            aria-hidden="true"
            class="la la-exchange-alt"
            role="presentation"
          />
        </span>
      </button>
    </div>
  </nav>
</DocumentFragment>
`;
